lib/open_project/scm/manageable_repository.rb

```diff
@@ -0,0 +1,115 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+module OpenProject
+  module Scm
+    module ManageableRepository
+      ##
+      # Reads from configuration whether repositories of this kind
+      # may be managed from OpenProject.
+      def manageable?
+        !managed_root.nil?
+      end
+
+      ##
+      # Returns the managed root for this repository vendor
+      def managed_root
+        scm.config[:manages]
+      end
+
+      ##
+      # Determines whether this repository IS managed
+      # by openproject
+      def managed?
+        managed
+      end
+
+      ##
+      # Allows descendants to perform actions
+      # with the given repository after the managed
+      # repository has been written to file system.
+      def managed_repo_created(args)
```

Comment:
Unused method argument - `args`. If it's necessary, use `_` or `_args` as an argument name to indicate that it won't be used. You can also write as `managed_repo_created(*)` if you want the method to accept any arguments but don't care about them.